src/main/java/bitbucketpullrequestbuilder/bitbucketpullrequestbuilder/BitbucketBuildTrigger.java

```diff
@@ -214,7 +216,16 @@ protected Job asJob() {
         return retrieveScheduleJob(this.job).scheduleBuild2(0,
                 new CauseAction(cause),
                 new ParametersAction(new ArrayList(values.values())),
-                new RevisionParameterAction(cause.getSourceCommitHash()));
+                new RevisionParameterAction(cause.getSourceCommitHash(), getBitbucketRepoUrl(cause.getRepositoryOwner(), cause.getRepositoryName())));
+    }
+
+    private URIish getBitbucketRepoUrl(String repoOwner, String repoName) {
+        try{
+            return new URIish(String.format("git@bitbucket.org:%s/%s.git", repoOwner, repoName));
```

Comment:
May want to grab the URL from the Git SCM configuration instead of partially hardcoding this path. Wasn't sure where to grab that from.

Reply:
Yeah, this is what I was trying to ask in the issue. I understand what you're trying to do, but, how do you get your url into the job's configuration?

Also, what if you need to pull from three urls? Where does the third come from.

While I understand the problem, for a generic fix, perhaps there are additional information you can add to your sub-repo fetch that allow it to work?

Reply:
Yea, they can be in any format git recognizes, which is ssh (which you have used) and different URL mechanisms (http, https, with or without authentication, etc)